vnext/Microsoft.ReactNative/ReactHost/CrashManager.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "CrashManager.h"
+#include "ReactInstanceWin.h"
+
+#include <winrt/Windows.Storage.h>
+
+#include <WerApi.h>
+
+namespace Mso::React {
+
+// when calling SetUnhandledExceptionFilter the previous filter is returned
+// this should be saved so that if needed it can be called after the new unhandled exception has finished executing
+// this allows the Windows Error Reporting system to process the error and upload the data for processing
+LPTOP_LEVEL_EXCEPTION_FILTER g_previousExceptionFilter = nullptr;
+
+std::wstring g_logFileName;
+bool g_wasCalled{false}; // Prevent recursion in the custom handler
+
+extern "C" LONG WINAPI CustomWERExceptionFilter(LPEXCEPTION_POINTERS const exceptionPointers) {
+  if (g_wasCalled) {
+    return EXCEPTION_EXECUTE_HANDLER;
+  }
+
+  g_wasCalled = true;
+
+  CrashManager::OnUnhandledException();
```

Comment:
> Cr [](http://example.com/codeflow?start=1&length=3)

(nit) should this be surrounded with try/catch or some other protection?

Reply:
I marked OnUnhandledException as noexcept just in case.